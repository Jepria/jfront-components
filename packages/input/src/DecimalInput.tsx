import React from "react"
import NumberFormat, { NumberFormatProps } from "react-number-format"
import styled from "styled-components"
import { Label } from "@jfront/ui-label"
import { LoadingImage, ExclamationImage } from "./styles"
import { InputProps } from "."

interface StyledNumberFormatProps {
  error?: string
}

const StyledNumberFormat = styled(NumberFormat)<StyledNumberFormatProps>`
  box-sizing: border-box;
  display: inline-block;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  text-align: left;
  height: 24px;
  ${(props) => (props.error ? "border: 1px solid red;" : "")};
  &:focus {
    ${(props) =>
      props.error
        ? "outline-color: red; outline-style: solid; outline-width: 1px;"
        : ""};
  }
`

export const DecimalInput = React.forwardRef<
  HTMLInputElement,
  InputProps & NumberFormatProps
>((props, ref) => {
  const { thousandSeparator = true, decimalSeparator = "." } = props

  return (
    <div>
      {props.label !== undefined && (
        <Label htmlFor={props.id}>{props.label}:&nbsp;</Label>
      )}
      <StyledNumberFormat
        {...props}
        thousandSeparator={thousandSeparator}
        decimalSeparator={decimalSeparator}
        getInputRef={ref}
      />
      {props.isLoading && <LoadingImage />}
      {props.error !== undefined && <ExclamationImage title={props.error} />}
    </div>
  )
})
