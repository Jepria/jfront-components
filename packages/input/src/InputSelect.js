var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
import React, { useState } from 'react';
export function InputSelect(props) {
    var list = [
        { id: "1", value: "Option1" },
        { id: "2", value: "Option2" },
        { id: "3", value: "Option3" }
    ]; // it could be func returning real data
    var _a = __read(useState(props.value || ""), 2), value = _a[0], setValue = _a[1];
    var _b = __read(useState(list), 2), data = _b[0], setData = _b[1];
    var _c = __read(useState(props.error || ""), 2), error = _c[0], setError = _c[1];
    var _d = __read(useState(props.disabled || false), 2), disabled = _d[0], setDisabled = _d[1];
    var _e = __read(useState(props.readonly || false), 2), readonly = _e[0], setEditable = _e[1];
    var handleChange = function (event) {
        alert(event.target.value);
        setValue(event.target.value);
    };
    var fieldClassName = ""; // make style depending on props and states
    var labelClassName = "label " + (error && "error");
    if (readonly) {
        return (React.createElement("div", { className: fieldClassName },
            React.createElement("label", { htmlFor: props.id, className: labelClassName },
                error || props.label,
                ":"),
            "\u00A0",
            value));
    }
    return (React.createElement("div", { className: fieldClassName },
        React.createElement("label", { htmlFor: props.id, className: labelClassName },
            error || props.label,
            ":"),
        "\u00A0",
        React.createElement("select", { id: props.id, name: props.name, value: value, disabled: disabled, onChange: handleChange.bind(this) }, data.map(function (item) { return (React.createElement("option", { key: item.id, value: item.value }, item.value)); }))));
}
//# sourceMappingURL=InputSelect.js.map