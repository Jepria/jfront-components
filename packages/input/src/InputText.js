var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
import React, { useState } from 'react';
export function InputText(props) {
    var _a = __read(useState((props.disabled && !props.readonly) || false), 2), active = _a[0], setActive = _a[1];
    var _b = __read(useState(props.value || ""), 2), value = _b[0], setValue = _b[1];
    var _c = __read(useState(props.error || ""), 2), error = _c[0], setError = _c[1];
    var _d = __read(useState(props.disabled || false), 2), disabled = _d[0], setDisabled = _d[1];
    var _e = __read(useState(props.readonly || false), 2), readonly = _e[0], setEditable = _e[1];
    function changeValue(event) {
        var value = event.target.value;
        setValue(value);
        setError("");
    }
    var fieldClassName = "field " + (active ? "active" : "");
    var labelClassName = "label " + (error && "error");
    return (React.createElement("div", { className: fieldClassName },
        React.createElement("label", { htmlFor: props.id, className: labelClassName },
            error || props.label,
            ":"),
        "\u00A0",
        React.createElement("input", { id: props.id, type: readonly ? "hidden" : "text", name: props.name, disabled: disabled, onChange: changeValue.bind(this), onFocus: function () { return !disabled && !readonly && setActive(true); }, onBlur: function () { return !disabled && !readonly && setActive(false); } }),
        readonly ? value : ""));
}
//# sourceMappingURL=InputText.js.map