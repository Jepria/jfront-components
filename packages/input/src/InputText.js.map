{"version":3,"file":"InputText.js","sourceRoot":"","sources":["InputText.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;AAAA,OAAO,KAAK,EAAE,EAAE,QAAQ,EAAE,MAAM,OAAO,CAAC;AAYxC,MAAM,UAAU,SAAS,CAAC,KAAqB;IAErC,IAAA,KAAA,OAAsB,QAAQ,CAAC,CAAC,KAAK,CAAC,QAAQ,IAAI,CAAC,KAAK,CAAC,QAAQ,CAAC,IAAI,KAAK,CAAC,IAAA,EAA3E,MAAM,QAAA,EAAE,SAAS,QAA0D,CAAC;IAC7E,IAAA,KAAA,OAAoB,QAAQ,CAAC,KAAK,CAAC,KAAK,IAAI,EAAE,CAAC,IAAA,EAA9C,KAAK,QAAA,EAAE,QAAQ,QAA+B,CAAC;IAChD,IAAA,KAAA,OAAoB,QAAQ,CAAC,KAAK,CAAC,KAAK,IAAI,EAAE,CAAC,IAAA,EAA9C,KAAK,QAAA,EAAE,QAAQ,QAA+B,CAAC;IAChD,IAAA,KAAA,OAA0B,QAAQ,CAAC,KAAK,CAAC,QAAQ,IAAI,KAAK,CAAC,IAAA,EAA1D,QAAQ,QAAA,EAAE,WAAW,QAAqC,CAAC;IAC5D,IAAA,KAAA,OAA0B,QAAQ,CAAC,KAAK,CAAC,QAAQ,IAAI,KAAK,CAAC,IAAA,EAA1D,QAAQ,QAAA,EAAE,WAAW,QAAqC,CAAC;IAElE,SAAS,WAAW,CAAC,KAAU;QAC3B,IAAM,KAAK,GAAG,KAAK,CAAC,MAAM,CAAC,KAAK,CAAC;QACjC,QAAQ,CAAC,KAAK,CAAC,CAAC;QAChB,QAAQ,CAAC,EAAE,CAAC,CAAC;IACjB,CAAC;IACD,IAAM,cAAc,GAAG,YAAS,MAAM,CAAC,CAAC,CAAC,QAAQ,CAAA,CAAC,CAAC,EAAE,CAAE,CAAC;IACxD,IAAM,cAAc,GAAG,YAAS,KAAK,IAAI,OAAO,CAAE,CAAC;IAC/C,OAAO,CACH,6BAAK,SAAS,EAAE,cAAc;QAC1B,+BAAO,OAAO,EAAE,KAAK,CAAC,EAAE,EAAE,SAAS,EAAE,cAAc;YAC9C,KAAK,IAAI,KAAK,CAAC,KAAK;gBACjB;;QACR,+BACI,EAAE,EAAE,KAAK,CAAC,EAAE,EACZ,IAAI,EAAE,QAAQ,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,CAAC,MAAM,EAClC,IAAI,EAAE,KAAK,CAAC,IAAI,EAChB,QAAQ,EAAE,QAAQ,EAClB,QAAQ,EAAE,WAAW,CAAC,IAAI,CAAC,IAAI,CAAC,EAChC,OAAO,EAAE,cAAM,OAAA,CAAC,QAAQ,IAAI,CAAC,QAAQ,IAAI,SAAS,CAAC,IAAI,CAAC,EAAzC,CAAyC,EACxD,MAAM,EAAE,cAAM,OAAA,CAAC,QAAQ,IAAI,CAAC,QAAQ,IAAI,SAAS,CAAC,KAAK,CAAC,EAA1C,CAA0C,GAC1D;QACD,QAAQ,CAAC,CAAC,CAAC,KAAK,CAAC,CAAC,CAAC,EAAE,CACpB,CACT,CAAC;AACV,CAAC"}