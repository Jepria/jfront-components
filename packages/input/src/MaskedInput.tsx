import React from "react"
import InputMask, { Props } from "react-input-mask"
import { TextInput } from "./TextInput"
import { InputProps } from "."

interface MaskPlaceholder {
  maskPlaceholder?: string
}

export const MaskedTextInput = React.forwardRef<
  HTMLInputElement,
  Props & InputProps & MaskPlaceholder
>((props, ref) => {
  return (
    <InputMask {...props} inputRef={ref}>
      <TextInput autoComplete="off" />
    </InputMask>
  )
})
