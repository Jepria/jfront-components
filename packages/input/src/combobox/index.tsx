import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import openIcon from './openIcon.gif';
import exclamation from '../images/exclamation.gif';
import loading from '../images/loading.gif';
import {useSelect, useFilter, UseFilterInstance, useDropdown, UseDropdownInstance} from "@jfront-ui/hooks";

export const ComboBoxInputContainer = styled.div`
  box-sizing: border-box;
  width: 100%;
  white-space: nowrap;
`;

export const ComboBoxButton = styled.img`
  box-sizing: border-box;
  display: inline-block;
  vertical-align: top;
  cursor: pointer;
  border-color: #ccc;
  border-top-color: #999;
  border-bottom: 1px solid #b5b8c8;
  height: 24px;
`;

export interface ComboBoxInputProps {
  error?: boolean;
}

export const ComboBoxInput = styled.input<ComboBoxInputProps>`
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  box-sizing: border-box;
  display: inline-block;
  vertical-align: top;
  height: 24px;
  padding-left: 2px;
  width: calc(100% - 17px);
  ${(props: ComboBoxInputProps) => props.error ? 'border: 1px solid red' : 'border: 1px solid #ccc; border-top: 1px solid #999;'}
`;

export interface ComboBoxOptionProps {
  selected?: boolean;
}

export const ComboBoxOption = styled.li<ComboBoxOptionProps>`
  overflow: hidden;
  white-space: nowrap;
  height: 18px;
  padding: 2px 6px;
  cursor: pointer;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  ${(props: ComboBoxOptionProps) => props.selected ? 'background: #ccddf3;' : '&:hover {background: #eee}'}
`;

export const ComboBoxList = styled.ul`
  background: white;
  overflow: auto;
  margin: 0;
  padding: 0;
  border-style: solid;
  border-color: #99BBE8;
  border-width: 1px;
  z-index: 5100;
  position: relative;
`;

interface ComboBoxProps {
  width?: string;
}

const ComboBox = styled.div`
  width: 100%;
`;

const Image = styled.img`
  height: 16px;
  width: 16px;
  margin-top: 2px;
  margin-left: 4px;
`;

export const Container = styled.div<ComboBoxProps>`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  min-width: ${props => props.width ? `calc(${props.width} / 2)` : '100px'};
  max-width: ${props => props.width ? props.width : '200px'};
  width: 100%;
`;


export interface ComboBoxFieldProps {
  id?: string;
  name?: string;
  initialValue?: any;
  touched?: boolean;
  error?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onChangeValue?: (field: string, value: any) => void;
  width?: string;
  options: Array<any>;
  isLoading?: boolean;
  getOptionName?: (option: any) => string;
  getOptionValue?: (option: any) => string;
  placeholder?: string;
  disabled?: boolean;
  hasEmptyOption?: boolean;
  emptyOptionText?: string;
}

/**
 * ComboBox form field
 *
 * @param {ComboBoxFieldProps} props incoming props
 * @example
 * const options = [
 *   {
 *     name: '123',
 *     value: '123'
 *   },
 *   {
 *     name: '111',
 *     value: '111'
 *   },
 *   {
 *     name: '222',
 *     value: '222'
 *   },
 *   {
 *     name: '333',
 *     value: '333'
 *   },
 *   {
 *     name: '444',
 *     value: '444'
 *   },
 * ]
 * <ComboBoxField options={options} onChangeValue={(field: string, value: any) => console.log(field, value)}/>
 */
export const ComboBoxField: React.FC<ComboBoxFieldProps> = ({
  id,
  width,
  name = '',
  initialValue,
  hasEmptyOption,
  emptyOptionText = '',
  touched,
  error,
  placeholder,
  disabled,
  onChange,
  onChangeValue,
  isLoading,
  options,
  getOptionName,
  getOptionValue }) => {

  const {
    isOpen,
    selectOption,
    getSelectedValue,
    getSelectedOption,
    getOptions,
    getButtonProps,
    getRootProps,
    getListProps,
    getInputProps
  } = useSelect({
    options: hasEmptyOption ? [{ name: emptyOptionText, value: undefined }, ...options] : [...options],
    onChange: (value) => {
      if (onChangeValue) {
        onChangeValue(name, value);
      }
    },
    initialValue,
    getOptionName,
    getOptionValue,
    invalidateOnFilter: true
  }, useDropdown, useFilter) as UseDropdownInstance & UseFilterInstance;

  const [filter, setFilter] = useState("");
  const [_isLoading, setIsLoading] = useState(isLoading);

  useEffect(
    () => {
      setIsLoading(isLoading);
    }, [isLoading]
  );

  const selectedValue = getSelectedValue();
  const selectedOption = getSelectedOption();

  return (
    <Container width={width}>
      <ComboBox {...getRootProps()} id={id}>
        <ComboBoxInputContainer>
          <ComboBoxInput
            {...getInputProps()}
            onChange={e => {
              if (onChange) {
                selectOption(undefined);
                setFilter(e.target.value);
                onChange(e);
              } else {
                getInputProps().onChange(e)
              }
            }}
            error={touched && error ? true : false}
            placeholder={placeholder}
            disabled={disabled}
            value={selectedOption ? (getOptionName ? getOptionName(selectedOption) : selectedOption.name) : (onChange ? filter : getInputProps().value)} />
          <ComboBoxButton {...getButtonProps()} src={openIcon} />
        </ComboBoxInputContainer>
        <ComboBoxList {...getListProps()} style={isOpen && !disabled ? { display: 'block' } : { display: 'none' }}>
          {
            getOptions().map(optionInstance => {
              return (
                <ComboBoxOption
                  {...optionInstance.getOptionProps()}
                  selected={selectedValue === (getOptionValue ? getOptionValue(optionInstance.option) : optionInstance.option?.value)}>
                  {getOptionName ? getOptionName(optionInstance.option) : optionInstance.option?.name}
                </ComboBoxOption>
              )
            })
          }
        </ComboBoxList>
      </ComboBox>
      {touched && error && <Image src={exclamation} title={error} />}
      {!error && _isLoading && <Image src={loading} />}
    </Container>
  );
}
