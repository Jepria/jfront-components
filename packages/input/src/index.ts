export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
