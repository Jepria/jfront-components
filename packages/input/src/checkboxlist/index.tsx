import React, { useReducer, createContext, useContext, useEffect, useState } from 'react';
import styled from 'styled-components';
import exclamation from '../images/exclamation.gif'
import loading from '../images/loading.gif';
import { useSelect, OptionProps } from '../../../hooks/src/useSelect';
import { useMultiple, UseMultipleInstance } from '../../../hooks/src/useSelect/useMultiple';

export const CheckBox = styled.input.attrs({ type: 'checkbox' })`
  margin: 2px;
`;

export const CheckBoxListOptionLabel = styled.label`
  height: 20px;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  display: inline-block;
  overflow: hidden;
  cursor: pointer;
  white-space: nowrap;
  width: 100%;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Safari */
   -khtml-user-select: none; /* Konqueror HTML */
     -moz-user-select: none; /* Old versions of Firefox */
      -ms-user-select: none; /* Internet Explorer/Edge */
          user-select: none; /* Non-prefixed version, currently
                                supported by Chrome, Edge, Opera and Firefox */
`;

export const CheckBoxListOption = styled.li`
  height: 20px;
  width: 100%;
  &:hover{
    background: #eee
  }
`;

export interface OptionListProps {
  error?: boolean;
}

export const OptionList = styled.ul<OptionListProps>`
  overflow: auto;
  background: white;
  margin: 0;
  padding: 0;
  width: 100%;
  height: calc(100% - 20px);
  ${props => props.error ? 'border: 1px solid red;' : 'border: 1px solid #ccc; border-top: 1px solid #999;'}
`;

export interface CheckBoxListProps {
  width?: string;
  height?: string;
}

export const CheckBoxList = styled.div<CheckBoxListProps>`
  box-sizing: border-box;
  padding: 0px;
  margin: 0px;
  font-family: tahoma, arial, helvetica, sans-serif;
  font-size: 12px;
  width: 100%;
  height: ${props => props.height ? props.height : '100px'};
`;

export const Icon = styled.img`
  margin-left: 5px;
  margin-top: 5px;
  height: 16px;
  width: 16px;
`;

export const ListContainer = styled.div<CheckBoxListProps>`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  min-width: ${props => props.width ? `calc(${props.width} / 2)` : '100px'};
  max-width: ${props => props.width ? props.width : '200px'};
  width: 100%;
`;

export interface CheckBoxListFieldProps {
  id?: string;
  name?: string;
  initialValue?: Array<any>;
  touched?: boolean;
  error?: string;
  isLoading?: boolean;
  width?: string;
  height?: string;
  onChangeValue?: (field: string, value: any) => void;
  options: Array<any>;
  getOptionName?: (option: any) => string;
  getOptionValue?: (option: any) => string;
  disabled?: boolean;
}

/**
 * Check box list form field
 *
 * @param {CheckBoxListFieldProps} props incoming props
 * @example
 * const options = [
 *   {
 *     name: '123',
 *     value: '123'
 *   },
 *   {
 *     name: '111',
 *     value: '111'
 *   },
 *   {
 *     name: '222',
 *     value: '222'
 *   },
 *   {
 *     name: '333',
 *     value: '333'
 *   },
 *   {
 *     name: '444',
 *     value: '444'
 *   },
 * ]
 * <CheckBoxListField options={options} onChangeValue={(field: string, value: any) => {console.log(field, value)}}/>
 */
export const CheckBoxListField: React.FC<CheckBoxListFieldProps> = ({
  id,
  name = '',
  initialValue,
  options,
  touched,
  error,
  isLoading,
  onChangeValue,
  getOptionName,
  getOptionValue,
  width,
  height }) => {

  const [_isLoading, setIsLoading] = useState(isLoading);

  useEffect(
    () => {
      setIsLoading(isLoading);
    }, [isLoading]
  );

  const {
    getSelectedValue,
    isSelectedAll,
    selectAll,
    selectOption,
    getOptions,
    getRootProps,
    getListProps
  } = useSelect({
    options: options,
    onChange: (value) => {
      if (onChangeValue) {
        onChangeValue(name, value);
      }
    },
    initialValue,
    getOptionName,
    getOptionValue,
  }, useMultiple
  ) as UseMultipleInstance;

  const value = getSelectedValue();

  return (
    <ListContainer width={width}>
      <CheckBoxList id={id} height={height} {...getRootProps()}>
        <OptionList error={touched && error ? true : false} {...getListProps()}>
          {
            getOptions().map(optionInstance => {
              return (<CheckBoxOption {...optionInstance.getOptionProps()} option={optionInstance.option} selected={value && value.includes(optionInstance.option.value)}/>)
            })
          }
        </OptionList>
        <SelectAllCheckBox selectAll={selectAll} selectOption={selectOption} isSelectedAll={isSelectedAll} disabled={!options || options.length == 0}/>
      </CheckBoxList>
      {touched && error && <Icon src={exclamation} title={error} />}
      {!error && _isLoading && <Icon src={loading} />}
    </ListContainer>
  );
}

export interface CheckBoxOptionProps extends OptionProps {
  selected: boolean;
  option: any;
  getOptionName?: (option: any) => string;
  getOptionValue?: (option: any) => string;
}

export const CheckBoxOption: React.FC<CheckBoxOptionProps> = (props) => {
  const { role, selected, option, getOptionName, onClick } = props;
  return (
    <CheckBoxListOption role={role}
      onDoubleClick={
        e => {
          /** IE fix checkbox double-click issue **/
          if ((document as any).documentMode) {
            onClick();
          }
        }
      } >
      <CheckBoxListOptionLabel>
        <CheckBox
          onChange={e => onClick()}
          checked={selected}
          onDoubleClick={
            e => {
              /** IE fix checkbox double-click issue **/
              if ((document as any).documentMode) {
                e.stopPropagation();
                onClick();
              }
            }
          }
        />
        {getOptionName ? getOptionName(option) : option.name}
      </CheckBoxListOptionLabel>
    </CheckBoxListOption>
  );
}

export interface SelectAllCheckBoxProps {
  disabled?: boolean;
  selectOption: (value: any) => void;
  selectAll: () => void;
  isSelectedAll: () => boolean;
}

export const SelectAllCheckBox: React.FC<SelectAllCheckBoxProps> = ({disabled, selectOption, selectAll, isSelectedAll}) => {

  const selectAllOptions = () => {
    if (isSelectedAll()) {
      selectOption([]);
    } else {
      selectAll();
    }
  }

  return (
    <CheckBoxListOptionLabel
      onDoubleClick={
        e => {
          /** IE fix checkbox double-click issue **/
          if ((document as any).documentMode) {
            selectAllOptions();
          }
        }
      }>
      <CheckBox
        onChange={() => {
          selectAllOptions();
        }}
        checked={isSelectedAll()}
        disabled={disabled}
        onDoubleClick={
          e => {
            /** IE fix checkbox double-click issue **/
            if ((document as any).documentMode) {
              e.stopPropagation();
              selectAllOptions();
            }
          }
        } />
          Выделить всё
    </CheckBoxListOptionLabel>
  );
}

