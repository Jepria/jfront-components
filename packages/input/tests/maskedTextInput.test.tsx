import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import { CheckBoxListInput, MaskedTextInput } from "../src"
import { AllByBoundAttribute } from "@testing-library/dom/types/queries"

test("Checking for the existence of an element MaskedTextInput", () => {
  render(<MaskedTextInput mask="99/99/9999" name="Masked" showMask />)
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <MaskedTextInput mask="99/99/9999" name="Masked" showMask />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Error display MaskedTextInput", () => {
  const { asFragment } = render(
    <MaskedTextInput mask="99/99/9999" name="Masked" showMask />,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Entering a value in MaskedTextInput", () => {
  let MaskedTextInputText
  render(
    <MaskedTextInput
      mask="99/99/9999"
      name="Masked"
      showMask
      data-testid="itemCheck"
      onChange={(name, value) => {
        MaskedTextInputText = value
      }}
      onFocus={(event) => {
        MaskedTextInputText = event.target.value
      }}
      value={123456}
      aria-label={"test-label"}
    />,
  )
  const input = screen.getByLabelText("test-label", { selector: "input" })
  fireEvent.focus(input)
  fireEvent.change(input, { target: { value: 99999999 } })
  fireEvent.focus(input)

  expect(MaskedTextInputText).toEqual("99/99/9999")
})
