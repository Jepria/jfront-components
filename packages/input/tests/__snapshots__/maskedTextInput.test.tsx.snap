// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error display MaskedTextInput 1`] = `
<DocumentFragment>
  <div
    class="sc-fzozJi jplEaj"
  >
    <input
      class="sc-fzoLsD gDlcHK"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-fzozJi jplEaj"
  >
    <input
      class="sc-fzoLsD gDlcHK"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
